### DISCUSSION TITLE
Facing issue in Velero backup - Backup Volumes:   <error getting backup volume info: gzip: invalid header>

### DISCUSSION THREAD

**Discussion by Shanky5554** [vmware-tanzu/velero#7689] [Community support Q&A]
Dear team, 
I am experiencing issues with Velero backups, encountering an error and also being unable to open the .gz files from the backups. My current setup involves using a Minio bucket for backup storage and Velero as the backup tool. Additionally, in our Kubernetes environment, we use Rook-Ceph as the storage file system. Could you please assist me in resolving these issues?

**Velero backup completed with error :-**
Phase:  Completed
Warnings:  error getting warnings: gzip: invalid header
Errors:  error getting errors: gzip: invalid header

<img width="355" alt="Screenshot 2024-04-16 at 5 41 08 PM" src="https://github.com/vmware-tanzu/velero/assets/167201458/8ae329d7-f8a6-4ee0-9015-312db61c6fd1">

**Comment by blackpiglet** [comment]
This means Velero CLI failed to download the backup result file from the backup repository. Please check whether you can download the backup's metadata files from the OSS manually.

**Reply by blackpiglet** [reply]
https://github.com/vmware-tanzu/velero/issues/7534
Please check whether this issue is related to your scenario.

**Reply by Shanky5554** [reply]
it is giving error when i create any backup plesae find debug log screenshot :-
<img width="1505" alt="Screenshot 2024-04-23 at 4 25 30 PM" src="https://github.com/vmware-tanzu/velero/assets/167201458/f33b8e0e-58e0-49f3-bc47-d7abef33d0be">

but i already have aws plugin installed in my code.

> ./velero plugin get -n minio-test
NAME                                            KIND
velero.io/crd-remap-version                     BackupItemAction
velero.io/crd-remap-version                     BackupItemAction
velero.io/pod                                   BackupItemAction
velero.io/pod                                   BackupItemAction
velero.io/pv                                    BackupItemAction
velero.io/pv                                    BackupItemAction
velero.io/service-account                       BackupItemAction
velero.io/service-account                       BackupItemAction
velero.io/csi-pvc-backupper                     BackupItemActionV2
velero.io/csi-volumesnapshot-backupper          BackupItemActionV2
velero.io/csi-volumesnapshotclass-backupper     BackupItemActionV2
velero.io/csi-volumesnapshotcontent-backupper   BackupItemActionV2
velero.io/csi-volumesnapshot-delete             DeleteItemAction
velero.io/csi-volumesnapshotcontent-delete      DeleteItemAction
velero.io/dataupload-delete                     DeleteItemAction
velero.io/aws                                   ObjectStore
velero.io/add-pv-from-pvc                       RestoreItemAction
velero.io/add-pv-from-pvc                       RestoreItemAction
velero.io/add-pvc-from-pod                      RestoreItemAction
velero.io/add-pvc-from-pod                      RestoreItemAction
velero.io/admission-webhook-configuration       RestoreItemAction
velero.io/admission-webhook-configuration       RestoreItemAction
velero.io/apiservice                            RestoreItemAction
velero.io/apiservice                            RestoreItemAction
velero.io/change-image-name                     RestoreItemAction
velero.io/change-image-name                     RestoreItemAction
velero.io/change-pvc-node-selector              RestoreItemAction
velero.io/change-pvc-node-selector              RestoreItemAction
velero.io/change-storage-class                  RestoreItemAction
velero.io/change-storage-class                  RestoreItemAction
velero.io/cluster-role-bindings                 RestoreItemAction
velero.io/cluster-role-bindings                 RestoreItemAction
velero.io/crd-preserve-fields                   RestoreItemAction
velero.io/crd-preserve-fields                   RestoreItemAction
velero.io/dataupload                            RestoreItemAction
velero.io/dataupload                            RestoreItemAction
velero.io/init-restore-hook                     RestoreItemAction
velero.io/init-restore-hook                     RestoreItemAction
velero.io/job                                   RestoreItemAction
velero.io/job                                   RestoreItemAction
velero.io/pod                                   RestoreItemAction
velero.io/pod                                   RestoreItemAction
velero.io/pod-volume-restore                    RestoreItemAction
velero.io/pod-volume-restore                    RestoreItemAction
velero.io/role-bindings                         RestoreItemAction
velero.io/role-bindings                         RestoreItemAction
velero.io/secret                                RestoreItemAction
velero.io/secret                                RestoreItemAction
velero.io/service                               RestoreItemAction
velero.io/service                               RestoreItemAction
velero.io/service-account                       RestoreItemAction
velero.io/service-account                       RestoreItemAction
velero.io/csi-pvc-restorer                      RestoreItemActionV2
velero.io/csi-volumesnapshot-restorer           RestoreItemActionV2
velero.io/csi-volumesnapshotclass-restorer      RestoreItemActionV2
velero.io/csi-volumesnapshotcontent-restorer    RestoreItemActionV2
velero.io/aws                                   VolumeSnapshotter

> 
also i added in my values.yaml file
<img width="816" alt="Screenshot 2024-04-23 at 4 28 43 PM" src="https://github.com/vmware-tanzu/velero/assets/167201458/0a679bfd-15e8-44d3-9bd1-20204e8cabd3">

create backup with below given command: 
> ./velero backup create webapp-1  --include-namespaces nginx-test --wait -n minio-test
Backup request "webapp-1" submitted successfully.
Waiting for backup to complete. You may safely press ctrl-c to stop waiting - your backup will continue in the background.
.......
Backup completed with status: Completed. You may check for more information using the commands `velero backup describe webapp-1` and `velero backup logs webapp-1`.

Output of backup 
<img width="637" alt="Screenshot 2024-04-23 at 4 30 06 PM" src="https://github.com/vmware-tanzu/velero/assets/167201458/48ca7f62-7012-457c-beb2-25d7d1ed40b8">

**Reply by blackpiglet** [reply]
Please collect the debug bundle by the command `velero debug`.
And please also give the Velero install method and details.

**Reply by blackpiglet** [reply]
Could you try with the Velero version of v1.13.2?
And the CSI plugin version is v0.7.1.

**Reply by Shanky5554** [reply]
sorry for the late reply...solution for me is i was giving minio-web URL in place of minio-api in publicURL option in velero config.....(I added an ingress at https://minio-api.example.com (obviously substituting example.com) pointing to the minio api port and changed my publicUrl in velero to use that and now both the logs and the resource list in the backup details is loading.) and downloaded file open through shell.
below ticket help me to get the solution.
https://github.com/vmware-tanzu/velero/issues/3900
Thanks for the support...!!!
